tests/integration/ef_tests/test_ef_blockchain_tests.py

```diff
@@ -67,6 +65,11 @@ async def setup_test_state(
                 starknet.state.state.cache._nonce_writes[starknet_address] = int(
                     account["nonce"], 16
                 )
+                await starknet.state.state.set_storage_at(
```

Comment:
It looks like you are only doing this for eoa's, but I think it needs to be done for all addresses, be it eoa or contract accounts.